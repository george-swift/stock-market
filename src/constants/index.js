export const API_BASE_URL = 'https://financialmodelingprep.com/api/v3/';

export const FETCH_ETF_LIST_STARTED = 'FETCH_ETF_LIST_STARTED';

export const FETCH_ETF_LIST_FAILED = 'FETCH_ETF_LIST_FAILED';

export const FETCH_ETF_LIST_SUCCEEDED = 'FETCH_ETF_LIST_SUCCEEDED';

export const FILTER_LISTING = 'FILTER_LISTING';

export const FILTER_NASDAQ = 'FILTER_NASDAQ';

export const FETCH_NASDAQ_STARTED = 'FETCH_NASDAQ_STARTED';

export const FETCH_NASDAQ_FAILED = 'FETCH_NASDAQ_FAILED';

export const FETCH_NASDAQ_SUCCEEDED = 'FETCH_NASDAQ_SUCCEEDED';

export const UI = { isLoading: false, error: null };

export const FILTER = { listing: '', nasdaq: '' };

export const sortByValue = (data) => data.sort((item1, item2) => item2.price - item1.price);
