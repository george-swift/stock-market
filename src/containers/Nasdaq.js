import '../assets/Sass/Listing.scss';
import { useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { FaSpinner } from 'react-icons/fa';
import { getFilteredNasdaq100 } from '../selectors';
import Constituent from '../components/NasdaqConstituent';
import { filterNasdaq } from '../actions';

const Nasdaq = () => {
  const companies = useSelector((state) => getFilteredNasdaq100(state));
  const { isLoading } = useSelector((state) => state.notifications);
  const dispatch = useDispatch();

  const handleSearch = (e) => dispatch(filterNasdaq(e.target.value));

  const navigate = useNavigate();

  const getUrl = (symbol) => navigate(`/company/${symbol}`);

  return (
    <div className="container-fluid px-0 listings">
      <div className="overview">
        <h2>List of NASDAQ 100 Companies</h2>
        <input
          type="search"
          className="form-control form-control-sm mb-3"
          placeholder="e.gs Technology, MSFT"
          onChange={handleSearch}
        />
      </div>
      {isLoading && <p className="page-loading"><FaSpinner /></p>}
      <table className="table table-hover bg-light">
        <thead className="table-dark">
          <tr>
            <th scope="col">Symbol</th>
            <th scope="col">Name</th>
            <th scope="col">Sector</th>
            <th scope="col">Headquarter</th>
            <th scope="col">Financial Model</th>
          </tr>
        </thead>
        <tbody>
          {
            companies?.map((company) => (
              <Constituent
                key={company.name}
                symbol={company.symbol}
                name={company.name}
                sector={company.sector}
                hq={company.headQuarter}
                getUrl={getUrl}
              />
            )) ?? (
              <tr>
                <th scope="row">n/a</th>
                <td colSpan="4">No companies listed in NASDAQ 100</td>
              </tr>
            )
          }
        </tbody>
      </table>
    </div>
  );
};

export default Nasdaq;
