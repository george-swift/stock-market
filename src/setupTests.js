/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });
