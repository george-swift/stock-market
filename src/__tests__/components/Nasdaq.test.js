import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import '../../setupTests';
import Nasdaq from '../../containers/Nasdaq';
import { FILTER, UI } from '../../constants';

const mockStore = configureStore();

describe('Snapshot of the Nasdaq component', () => {
  const store = mockStore({
    nasdaq: [
      {
        symbol: 'CHTR',
        name: 'Charter Communications Inc',
        sector: 'Communication Services',
        headQuarter: 'Stamford, CONNECTICUT',
      },
      {
        symbol: 'VRSN',
        name: 'Verisign Inc',
        sector: 'Technology',
        headQuarter: 'Reston, VIRGINIA',
      },
      {
        symbol: 'SPLK',
        name: 'Splunk Inc',
        sector: 'Technology',
        headQuarter: 'San Francisco, CALIFORNIA',
      },
    ],
    filter: { ...FILTER },
    notifications: { ...UI },
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <Nasdaq />
        </Router>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Constituent').exists()).toBe(true);
  });
});

describe('Querying the Nasdaq component', () => {
  it('should render a spinning component while the page loads', () => {
    const store = mockStore({
      nasdaq: [],
      filter: { ...FILTER },
      notifications: {
        ...UI,
        isLoading: true,
      },
    });

    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <Nasdaq />
        </Router>
      </Provider>,
    );

    expect(wrapper.find('FaSpinner').exists()).toBe(true);
    expect(wrapper.find('Constituent').exists()).toBe(false);
  });
});
