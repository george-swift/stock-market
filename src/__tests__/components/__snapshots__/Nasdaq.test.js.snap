// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Nasdaq component should render correctly 1`] = `ReactWrapper {}`;
